src/main/java/chatbot/task/TaskManager.java

```diff
@@ -0,0 +1,216 @@
+package chatbot.task;
+
+import chatbot.ChatbotException;
+import chatbot.storage.Storage;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * The TaskManager class manages tasks, including adding, marking, unmarking, printing, and deleting tasks.
+ */
+public class TaskManager {
+    private TaskList tasks;
+    private Storage storage;
+
+
+    /**
+     * Constructor for this class.
+     */
+    public TaskManager() {
+        this.tasks = new TaskList();
+        this.storage = new Storage("./data/chatbot.txt");
+        this.tasks = storage.loadFromFile();
+    }
+
+    /**
+     * Add todo task into the list.
+     *
+     * @param t description of todo
+     * @throws ChatbotException when the description is empty
+     */
+    public void addTodo(String t) throws ChatbotException {
+        if (t == null || t.trim().isEmpty()) {
+            throw new ChatbotException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        Task task = new Todos(t);
+        tasks.add(task);
+        storage.saveToFile(tasks);
+        System.out.println("    ____________________________________________________________");
```

Comment:
Maybe you could consider having a static method like "linePrinter" in another UI class that can be called to print this line of dashes instead of manually writing them?